pkg/platform/runtime/buildscript/file.go

```diff
@@ -0,0 +1,113 @@
+package buildscript
+
+import (
+	"bytes"
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/pkg/platform/api/graphql/model/buildplanner"
+
+	"github.com/alecthomas/participle/v2"
+	"gopkg.in/yaml.v2"
+)
+
+type DoesNotExistError struct{ error }
+
+func IsDoesNotExistError(err error) bool {
+	return errs.Matches(err, &DoesNotExistError{})
+}
+
+type File struct {
+	Path   string
+	Script *Script
+}
+
+func get(path string) (*File, error) {
```

Comment:
I'll make `NewScript()`, but I don't see the utility in `NewFile()` because we're only supposed to have one build script per project, like `project.Get()`.

Reply:
Maybe @Naatan will correct me... Get is a workaround for legacy reliance on globals. We intend to drop that in favor of structuring dependencies throughout the application. That is, project.Project would be called higher in the stack, and passed around where it's needed. Keep in mind that Get does forward to project.New.

The official docs suggest the structuring approach, and also to avoid the "get" term in naming: https://go.dev/doc/effective_go#package-names. It's worth adding that the docs recommend a more package-oriented approach to design than we tend to take at this time. It does result in more files and packages, but the code tends to be more modular.

Reply:
Get() in the project package is definitely legacy functionality, because it relies on globals. Seems this isn't relying on globals though, so I don't see the issue. We don't have a ban on the word "get", just on globals.

Reply:
Very well, I'll use NewFile as well then.